core/src/main/scala/org/broadinstitute/dsde/rawls/submissions/SubmissionsService.scala

```diff
@@ -612,15 +613,36 @@ class SubmissionsService(
       header = SubmissionValidationHeader(methodConfig.rootEntityType, methodConfigInputs, entityProvider.entityStoreId)
 
       workspaceExpressionResults <- evaluateWorkspaceExpressions(workspaceContext, gatherInputsResult)
-      submissionParameters <- entityProvider.evaluateExpressions(
-        ExpressionEvaluationContext(submissionRequest.entityType,
-                                    submissionRequest.entityName,
-                                    submissionRequest.expression,
-                                    methodConfig.rootEntityType
-        ),
-        gatherInputsResult,
-        workspaceExpressionResults
-      )
+      submissionParameters <-
+        if (submissionRequest.entityNames.isEmpty) {
```

Comment:
Some fairly ugly logic in this file to determine whether we have single or multiple entities; it could probably be smartened up a bit but some amount of it is unavoidable if we stick with adding an extra column

Reply:
Could you do some early logic to inspect both `entityNames` and `entityName` to calculate the input before doing any of the evaluation? Something like:
```scala
      // if the user supplied entityNames, use those; else, turn the single entityName into a Seq and
      // pretend it was supplied as entityNames
      val entityNamesOnWhichToRunTheSubmission: Option[Seq[String]] = submissionRequest.entityNames.
        getOrElse(submissionRequest.entityName.map(Seq(_)))
```
Once you've normalized `entityNames` and `entityName` into the same structure, you can just use that structure for evaluation.